### DISCUSSION TITLE
How to handle types defined in different assemblies

### DISCUSSION THREAD

**Discussion by trampster** [YairHalberstadt/stronginject#119] [Q&A]
We are using Xamarin.Forms so we have .net standard shared project, with most of our types defined, but also we have some services which must be defined in platform specific projects.

The platform specific services have interfaces which are defined in the shared project, that way shared services can depend on the interfaces and the implementation can be in the plaform specific project.

(there is a xamarin example but it only has shared services so only has one container defined in the shared project)

Ideally we would like to be able to define a container in the shared project with all the shared services and a container in the platform specific project containing the platform specific services.

Is it possible to define multiple containers in different projects and have them work together so that the shared project container can resolve services defined in the platform specific container?

**Comment by trampster** [comment]
I've discovered Modules which seems like they could fix the problem but actually doesn't.

I can define two modules one in the shared project and one in the platform project.
Then I can define a container in the platform project.
However I then can't resolve a type from the shared project (App.xaml.cs) because the container is defined in the platform project which I don't have a reference to.

**Reply by YairHalberstadt** [reply]
You could use the container interface to resolve the type in the shared project.

**Reply by trampster** [reply]
Thanks, looks like you need an interface per type you need to resolve, but I've got it working by defining a interface that implements all the types I need.
